controllers/video.controller.js

```diff
@@ -0,0 +1,38 @@
+const {Video}=require("../models/video-model");
+
+const getVideos=async(req,res)=>{
+  try{
+    const videos = await Video.find({});
+    res.json({success:true,videos})
+  }catch(err){
+    res.status(500).json({
+      success:false,
+      message:"Unable to get videos",
+      errMessage:err.message
+    })
+  }
+};
+
+const findVideo=async(req,res,next,vid)=>{
+  try{
+    const video=await Video.findById(vid);
+    if(!video){
```

Comment:
You can make "vid" an index in Schema for a faster search.

Reply:
Will see into it